LeanHelix panic

```
panic: Accessed union field of incorrect type, did you check which union type it is first?

goroutine 1 [running]:
github.com/orbs-network/orbs-network-go/vendor/github.com/orbs-network/orbs-spec/types/go/protocol.(*TransactionsBlockProof).LeanHelix(0xc000397980, 0xc000368770, 0x2, 0x2)
	/go/src/github.com/orbs-network/orbs-network-go/vendor/github.com/orbs-network/orbs-spec/types/go/protocol/blocks.mb.go:982 +0x1d6
github.com/orbs-network/orbs-network-go/services/consensusalgo/leanhelixconsensus.ExtractBlockProof(0xc000368db0, 0xe5a3ce, 0x43, 0xc000368770)
	/go/src/github.com/orbs-network/orbs-network-go/services/consensusalgo/leanhelixconsensus/service.go:183 +0x32
github.com/orbs-network/orbs-network-go/services/consensusalgo/leanhelixconsensus.(*service).HandleBlockConsensus(0xc00029a2d0, 0xf266e0, 0xc00008c018, 0xc00000c3c0, 0x0, 0x0, 0x0)
	/go/src/github.com/orbs-network/orbs-network-go/services/consensusalgo/leanhelixconsensus/service.go:171 +0x1d9
github.com/orbs-network/orbs-network-go/services/blockstorage.(*service).notifyConsensusAlgos(0xc000096a00, 0xf266e0, 0xc00008c018, 0x0, 0xc000368db0, 0x2, 0x0, 0x0)
	/go/src/github.com/orbs-network/orbs-network-go/services/blockstorage/validate_block.go:135 +0x17e
github.com/orbs-network/orbs-network-go/services/blockstorage.(*service).UpdateConsensusAlgosAboutLastCommittedBlockInLocalPersistence(0xc000096a00, 0xf266e0, 0xc00008c018)
	/go/src/github.com/orbs-network/orbs-network-go/services/blockstorage/block_sync_bridge.go:26 +0x138
github.com/orbs-network/orbs-network-go/services/blockstorage.(*service).RegisterConsensusBlocksHandler(0xc000096a00, 0xf1bba0, 0xc00029a2d0)
	/go/src/github.com/orbs-network/orbs-network-go/services/blockstorage/init.go:101 +0x65
github.com/orbs-network/orbs-network-go/services/consensusalgo/leanhelixconsensus.NewLeanHelixConsensusAlgo(0xf266a0, 0xc000094bc0, 0x7f39f1d40718, 0xc0000e8370, 0xf2b100, 0xc000096a00, 0xf289a0, 0xc0000938c0, 0xf2a440, 0xc000092c60, ...)
	/go/src/github.com/orbs-network/orbs-network-go/services/consensusalgo/leanhelixconsensus/service.go:134 +0xba1
github.com/orbs-network/orbs-network-go/bootstrap.NewNodeLogic(0xf266a0, 0xc000094bc0, 0xf1e720, 0xc000397c20, 0xf29600, 0xc000082140, 0xf25420, 0xc000082370, 0x0, 0x0, ...)
	/go/src/github.com/orbs-network/orbs-network-go/bootstrap/node_logic.go:72 +0xd59
github.com/orbs-network/orbs-network-go/bootstrap.NewNode(0x7f39f1d2d610, 0xc0000b3340, 0xf2a440, 0xc000092c00, 0x0, 0x0)
	/go/src/github.com/orbs-network/orbs-network-go/bootstrap/node.go:56 +0x57f
main.main()
	/go/src/github.com/orbs-network/orbs-network-go/main.go:107 +0x38c
```

--- a/services/consensusalgo/leanhelixconsensus/service.go
+++ b/services/consensusalgo/leanhelixconsensus/service.go
@@ -169,7 +169,11 @@ func (s *service) HandleBlockConsensus(ctx context.Context, input *handlers.Hand
 			s.logger.Info("HandleBlockConsensus(): Calling UpdateState in LeanHelix with GenesisBlock", log.Stringable("mode", input.Mode))
 		} else { // we should have a lhBlock proof
 			s.logger.Info("HandleBlockConsensus(): Calling UpdateState in LeanHelix with block", log.Stringable("mode", input.Mode), log.BlockHeight(blockPair.TransactionsBlock.Header.BlockHeight()))
-			lhBlockProof = ExtractBlockProof(blockPair)
+			var err error
+			lhBlockProof, err = ExtractBlockProof(blockPair)
+			if err != nil {
+				return nil, err
+			}
 			lhBlock = ToLeanHelixBlock(blockPair)
 
 		}
@@ -180,8 +184,14 @@ func (s *service) HandleBlockConsensus(ctx context.Context, input *handlers.Hand
 	return nil, nil
 }
 
-func ExtractBlockProof(blockPair *protocol.BlockPairContainer) primitives.LeanHelixBlockProof {
-	return blockPair.TransactionsBlock.BlockProof.LeanHelix()
+func ExtractBlockProof(blockPair *protocol.BlockPairContainer) (primitives.LeanHelixBlockProof, error) {
+	if blockPair == nil || blockPair.TransactionsBlock == nil || blockPair.TransactionsBlock.BlockProof == nil {
+		return nil, errors.New("blockPair or TransactionsBlock or BlockProof is nil")
+	}
+	if !blockPair.TransactionsBlock.BlockProof.IsTypeLeanHelix() {
+		return nil, errors.New("BlockProof is not of type LeanHelix")
+	}
+	return blockPair.TransactionsBlock.BlockProof.LeanHelix(), nil
 }
 
 func (s *service) HandleLeanHelixMessage(ctx context.Context, input *gossiptopics.LeanHelixInput) (*gossiptopics.EmptyOutput, error) {

--- a/services/consensusalgo/leanhelixconsensus/test/block_proof_serialization_test.go
+++ b/services/consensusalgo/leanhelixconsensus/test/block_proof_serialization_test.go
@@ -5,6 +5,8 @@ import (
 	lhbuilders "github.com/orbs-network/lean-helix-go/test/builders"
 	"github.com/orbs-network/orbs-network-go/services/consensusalgo/leanhelixconsensus"
 	"github.com/orbs-network/orbs-network-go/test/builders"
+	"github.com/orbs-network/orbs-spec/types/go/protocol"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/consensus"
 	"github.com/stretchr/testify/require"
 	"testing"
 )
@@ -13,7 +15,8 @@ func TestBlockProofSerialization(t *testing.T) {
 	expectedBlockProof := lhbuilders.AMockBlockProof().Raw()
 	block := builders.BlockPair().Build()
 	block.TransactionsBlock.BlockProof = leanhelixconsensus.CreateTransactionBlockProof(block, expectedBlockProof)
-	actualBlockProof := leanhelixconsensus.ExtractBlockProof(block)
+	actualBlockProof, err := leanhelixconsensus.ExtractBlockProof(block)
+	require.NoError(t, err, "ExtractBlockProof should succeed if block contains LeanHelix proof")
 	require.True(t, bytes.Equal(expectedBlockProof, actualBlockProof), "block proofs should be the same: expectedBlockProof=%v actualBlockProof=%v", expectedBlockProof, actualBlockProof)
 }
 
@@ -23,4 +26,14 @@ func TestBlockProofSerialization_IncorrectExtraction(t *testing.T) {
 	block.TransactionsBlock.BlockProof = leanhelixconsensus.CreateTransactionBlockProof(block, expectedBlockProof)
 	incorrectBlockProof := block.TransactionsBlock.BlockProof.Raw()
 	require.False(t, bytes.Equal(expectedBlockProof, incorrectBlockProof), "block proofs should be different")
+}
+
+func TestExtractBlockProofFailsIfNotLeanHelixBlock(t *testing.T) {
+	block := builders.BlockPair().Build()
+	block.TransactionsBlock.BlockProof = (&protocol.TransactionsBlockProofBuilder{
+		Type:               protocol.TRANSACTIONS_BLOCK_PROOF_TYPE_BENCHMARK_CONSENSUS,
+		BenchmarkConsensus: &consensus.BenchmarkConsensusBlockProofBuilder{},
+	}).Build()
+	_, err := leanhelixconsensus.ExtractBlockProof(block)
+	require.Error(t, err, "ExtractBlockProof should fail if block does not contain LeanHelix proof")
 }

